### DISCUSSION TITLE
Orthogonal reparameterization for Dense layer in Flax/Jax

### DISCUSSION THREAD

**Discussion by Chutlhu** [google/flax#3626] [Q&A]
Dear all,
is there an equivalent of the following pytorch reparameterization in Flax?
https://pytorch.org/docs/stable/generated/torch.nn.utils.parametrizations.orthogonal.html

**Comment by davisyoshida** [comment]
This would be pretty easy to implement using [https://github.com/davisyoshida/qax](Qax), a tool I made to make writing this class of transformation easier.

**Comment by chiamp** [comment]
Flax does not have a specific orthogonal parametrization function, but we have [`nn.map_variables`](https://flax.readthedocs.io/en/latest/api_reference/flax.linen/_autosummary/flax.linen.map_variables.html), which is the Flax equivalent of PyTorch's [`parametrize`](https://pytorch.org/tutorials/intermediate/parametrizations.html)
